lib/source-code/source-code.js

```diff
@@ -724,6 +895,184 @@ class SourceCode extends TokenStore {
+            languageOptions.globals
+        );
+        const varsCache = this[caches].get("vars");
+
+        varsCache.set("configGlobals", configGlobals);
+    }
+
+    /**
+     * Applies configuration found inside of the source code. This method is only
+     * called when ESLint is running with inline configuration allowed.
+     * @returns {{ok:boolean,problems:Array<Problem>,configs:{config:FlatConfigArray,node:ASTNode}}} Information
+     *      that ESLint needs to further process the inline configuration.
+     */
+    applyInlineConfig() {
+
+        const problems = [];
+        const configs = [];
+        const exportedVariables = {};
+        const inlineGlobals = Object.create(null);
+
+        this.getInlineConfigNodes().forEach(comment => {
+
+            const { directivePart } = extractDirectiveComment(comment.value);
+            const match = directivesPattern.exec(directivePart);
+            const directiveText = match[1];
+            const directiveValue = directivePart.slice(match.index + directiveText.length);
+
+            switch (directiveText) {
+                case "exported":
+                    Object.assign(exportedVariables, commentParser.parseStringConfig(directiveValue, comment));
+                    break;
+
+                case "globals":
+                case "global":
+                    for (const [id, { value }] of Object.entries(commentParser.parseStringConfig(directiveValue, comment))) {
+                        let normalizedValue;
+
+                        try {
+                            normalizedValue = normalizeConfigGlobal(value);
+                        } catch (err) {
+                            problems.push({
+                                ruleId: null,
+                                loc: comment.loc,
+                                message: err.message
+                            });
+                            continue;
+                        }
+
+                        if (inlineGlobals[id]) {
+                            inlineGlobals[id].comments.push(comment);
+                            inlineGlobals[id].value = normalizedValue;
+                        } else {
+                            inlineGlobals[id] = {
+                                comments: [comment],
+                                value: normalizedValue
+                            };
+                        }
+                    }
+                    break;
+
+                case "eslint": {
+                    const parseResult = commentParser.parseJsonConfig(directiveValue, comment.loc);
+
+                    if (parseResult.success) {
+                        configs.push({
+                            config: {
+                                rules: parseResult.config
+                            },
+                            node: comment
+                        });
+                    } else {
+                        problems.push(parseResult.error);
```

Comment:
This linting problem will be different than before this change because it wasn't going through `createLintingProblem()`.

Before:

```json
{
  "ruleId": null,
  "fatal": true,
  "severity": 2,
  "message": "Failed to parse JSON from ' foo': Unexpected token f in JSON at position 2",
  "line": 1,
  "column": 1,
  "nodeType": null
}
```

After:

```json
{
  "ruleId": null,
  "message": "Failed to parse JSON from ' foo': Unexpected token f in JSON at position 2",
  "line": 1,
  "column": 1,
  "endLine": 1,
  "endColumn": 2,
  "severity": 2,
  "nodeType": null
}
```

A notable difference is `"fatal": true` that would be omitted.

Reply:
Ah good call. I'll add that.

Reply:
So it looks like there's an inconsistency within eslintrc mode where some inline config problems have `fatal: true` (parsing rule configs) but others don't (invalid `global` values). For simplicity, I added `fatal: true` to all of these in flat config mode.